location.conf

```diff
@@ -0,0 +1,58 @@
+root /home/app/webapp/public;
+
+include /usr/local/openresty/nginx/conf/pagespeed.conf;
+
+location = /maintenance.html {
+    root /home/app;
+}
+
+set $thescheme 'http';
+if ($http_x_forwarded_proto = 'https') {
+  set $thescheme 'https';
+}
+if ($scheme = 'https') {
+  set $thescheme 'https';
+}
+
+location /monitors/lb {
+    access_by_lua_file /home/app/maintenance.lua;
+
+    try_files $uri @app;
+}
+
+location / {
+    rewrite_by_lua_file /home/app/redirect.lua;
+    access_by_lua_file /home/app/maintenance.lua;
+
+
+    if ($thescheme = 'http') {
+      return 301 https://$host$request_uri;
+    }
+
+    try_files $uri $uri/index.html @app;
+}
+
+location @app {
+    proxy_pass 'http://puma';
+    proxy_http_version 1.1;
+    proxy_read_timeout 600;
+    proxy_set_header Connection "";
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header Host $host;
+    proxy_set_header X-Real-IP $remote_addr;
+    proxy_set_header X-Forwarded-Proto $thescheme;
+    proxy_set_header X-Amzn-Trace-Id $http_x_amzn_trace_id;
+    proxy_redirect off;
+    proxy_set_header X-Request-Start "t=${msec}";
+}
+
+location /cable {
+    proxy_pass http://puma;
+    proxy_http_version 1.1;
+    proxy_set_header Upgrade $http_upgrade;
+    proxy_set_header Connection "upgrade";
+}
+
```

Comment:
The net effect is to add these six lines to the file in ecs_config/ecs/base-image-ruby-version-arg

The danger is that when ecs_config is updated, we are likely to miss those updates here. For example, `voke_api` and `watchthinkchat` have both drifted in this fashion.

It may be wiser to add a script to the build that edits these lines into the existing location.conf, which could be as simple as appending them to the end with `cat`. Thoughts, please, @frett @Omicron7 @jbirdjavi ?

Reply:
I consider concatenation of config files like this a bad practice. If you'd really like to allow apps to override sections, or add pieces here, we should add and `include` directive to the base image, and let apps create additional .conf files that get included.

Reply:
> we should add an include directive to the base image

That would be nice, so long as it doesn't error when there is no file to include (the default case). Or, I suppose, the include file could be present in ecs_config but empty.

Reply:
Something like
```
include /usr/local/openresty/nginx/locations/*.conf;
```
Anywhere `location.conf` is included. Then you copy your `cable.conf` to that location in the Dockerfile and nginx will pull it in.
https://github.com/CruGlobal/ecs_config/blob/master/ecs/base-image-openresty/nginx-httpd.conf#L28

Reply:
I like the idea of supporting additional locations through an include. it would let us fully leverage the base config but still provide overrides when necessary

Reply:
@Omicron7 @frett I created https://github.com/CruGlobal/ecs_config/pull/634/files